org.lflang/src/org/lflang/generator/ts/TsGenerator.kt

```diff
@@ -0,0 +1,436 @@
+            val (mainParameters, parameterCode) = parameterGenerator.generatePrameters()
+            tsCode.append(parameterCode)
+
+            val reactorGenerator = TsReactorGenerator(this, errorReporter)
+            for (reactor in reactors) {
+                reactorGenerator.generateReactor(reactor, federate)
+            }
+            reactorGenerator.generateReactorInstanceAndStart(this.mainDef, mainParameters)
+            tsCode.append(reactorGenerator.getCode())
+            fsa.generateFile(fileConfig.srcGenBasePath.relativize(tsFilePath).toString(),
+                tsCode.toString())
+        }
+
+        // Run necessary commands.
+
+        val pnpmInstall = commandFactory.createCommand(
+            "pnpm",
+            listOf("install"),
+        fileConfig.getSrcGenPkgPath(),
+        false // only produce a warning if command is not found
+        )
+
+        // Attempt to use pnpm, but fall back on npm if it is not available.
+        if (pnpmInstall !== null) {
+            val ret = pnpmInstall.run()
+            if (ret !== 0) {
+                errorReporter.reportError(findTarget(resource),
+                    "ERROR: pnpm install command failed: " + pnpmInstall.errors.toString())
+            }
+        } else {
+            errorReporter.reportWarning(
+                "Falling back on npm. To prevent an accumulation of replicated dependencies, " +
+                        "it is highly recommended to install pnpm globally (npm install -g pnpm).")
+            val npmInstall = commandFactory.createCommand("npm", listOf("install"), fileConfig.getSrcGenPkgPath())
+
+            if (npmInstall === null) {
+                errorReporter.reportError(
+                    "The TypeScript target requires npm >= 6.14.4. " +
+                            "For installation instructions, see: https://www.npmjs.com/get-npm. \n" +
+                            "Auto-compiling can be disabled using the \"no-compile: true\" target property.")
+                return
+            }
+
+            if (npmInstall.run() !== 0) {
+                errorReporter.reportError(findTarget(resource),
+                    "ERROR: npm install command failed: " + npmInstall.errors.toString())
+                errorReporter.reportError(findTarget(resource), "ERROR: npm install command failed." +
+                        "\nFor installation instructions, see: https://www.npmjs.com/get-npm")
+                return
+            }
+        }
+
+        refreshProject()
+
+        // Invoke the protocol buffers compiler on all .proto files in the project directory
+        // Assumes protoc compiler has been installed on this machine
+
+        // First test if the project directory contains any .proto files
+        if (targetConfig.protoFiles.size != 0) {
+            // For more info, see: https://www.npmjs.com/package/ts-protoc-gen
+
+            // FIXME: Check whether protoc is installed and provides hints how to install if it cannot be found.
```

Comment:
We should start use @cmnrd's `LFCommand` class here.

Reply:
I'm not sure if I understood your comment correctly. It looks like we're already using LFCommand (created by commandFactory.createCommand() below). Could you clarify a bit?

Reply:
Oh, you're right. In that case the `FIXME` on that line must be obsolete.